app/models/ems_refresh/save_inventory_physical_infra.rb

```diff
@@ -58,4 +59,35 @@ def save_asset_details_inventory(parent, hash)
     return if hash.nil?
     save_inventory_single(:asset_details, parent, hash)
   end
+
+  def ipaddress?(hostname)
+    IPAddr.new(hostname)
+    return true
+  rescue
+    return false
+  end
+
+  def resolve_hostname(ipaddress, ems)
+    ems.hostname = "https://#{Resolv.getname(ipaddress)}/"
+    _log.info("EMS ID: #{ems.id}" + " Resolved hostname successfully.")
+  rescue => err
+    _log.warn("EMS ID: #{ems.id}" + " It's not possible resolve hostname of the physical infra, #{err}.")
+  end
+
+  def resolve_ip_address(hostname, ems)
+    ems.ipaddress = Resolv.getaddress(hostname)
```

Comment:
@blomquisg yes https://github.com/ManageIQ/manageiq/pull/15252 added the ability to update the ems in save_ems_inventory
And https://github.com/ManageIQ/manageiq-providers-vmware/pull/62 is an (unmerged :wink: :wink:) example of its use.